### ISSUE TITLE
Make `--fast` meaningful again

### ISSUE THREAD

**Issue by bconway** [golangci/golangci-lint#1810]
Problem: When every linter claims it's fast, "fast" no longer has meaning:

```
$ golangci-lint linters|grep 'fast: true'|wc -l
      68

$ golangci-lint linters|grep 'fast: false'
unused (megacheck): Checks Go code for ... [fast: false, auto-fix: false]
```

I don't want to pick on anyone's work, but many of the linters are in fact quite slow. I wouldn't use many of them with IDE on-save functionality, anyway.

A couple ideas:
- Get rid of the "fast" categorization and get rid of `--fast`. This is probably not a good proposal.
- Use a quantitative metric to decide what is "fast." Not sure what this would look like or when/where it would be performed.

Thanks for the consideration.

**Comment by ldez** [comment]
I created some benchmarks on the code of `golangci-lint` and `traefik`.
The script resets the cache before each command execution.

<details>
<summary>golangci-lint</summary>

![prefs_golangci-lint](https://user-images.githubusercontent.com/5674651/111066139-c85aa080-84bd-11eb-977b-d84a28710749.png)

</details>

<details>
<summary>traefik</summary>

![perfs_traefik](https://user-images.githubusercontent.com/5674651/111066138-c7c20a00-84bd-11eb-8de2-d1f1beee0433.png)

</details>

<details>
<summary>the script</summary>

```bash
#!/bin/bash

linters=(
	"asciicheck"
	"bodyclose"
	"cyclop"
	"deadcode"
	"depguard"
	"dogsled"
	"dupl"
	"durationcheck"
	"errcheck"
	"errorlint"
	"exhaustive"
	"exhaustivestruct"
	"exportloopref"
	"forbidigo"
	"forcetypeassert"
	"funlen"
	"gci"
	"gochecknoglobals"
	"gochecknoinits"
	"gocognit"
	"goconst"
	"gocritic"
	"gocyclo"
	"godot"
	"godox"
	"goerr113"
	"gofmt"
	"gofumpt"
	"goheader"
	"goimports"
	"golint"
	"gomnd"
	"gomodguard"
	"goprintffuncname"
	"gosec"
	"gosimple"
	"govet"
	"ifshort"
	"importas"
	"ineffassign"
#	"interfacer"
	"lll"
	"makezero"
#	"maligned"
	"misspell"
	"nakedret"
	"nestif"
	"nilerr"
	"nlreturn"
	"noctx"
	"nolintlint"
	"paralleltest"
	"prealloc"
	"predeclared"
	"revive"
	"rowserrcheck"
	"scopelint"
	"sqlclosecheck"
	"staticcheck"
	"structcheck"
	"stylecheck"
	"testpackage"
	"thelper"
	"tparallel"
	"typecheck"
	"unconvert"
	"unparam"
	"unused"
	"varcheck"
	"wastedassign"
	"whitespace"
	"wrapcheck"
	"wsl"
)

echo 'name,real,user,sys'

for item in "${linters[@]}"
do
  golangci-lint cache clean
  printf "$item,"
  { /usr/bin/time -q -f"%e,%U,%S" golangci-lint run --no-config --disable-all --print-issued-lines=false -E$item > /dev/null; } 2>&1
  echo
done
```

</details>

**Comment by bconway** [comment]
Wow, great work! Thanks for diving in on that. I see your point on the nuances of a `fast` boolean.

One nice advantage to having such a toggle is that I can minimize configuration by "using" the same set of tests in a Makefile/CI and in my IDE, with the addition of the `fast` flag for interactivity. Obviously "using" is in quotes here, as not all the same tests are run.

**Comment by ldez** [comment]
I analyzed the code related to the status `fast/slow`:

`IsSlowLinter` define if a linter is slow, but `lc.LoadMode&packages.NeedTypesInfo != 0` is always `false` because we never use this load mode.

https://github.com/golangci/golangci-lint/blob/92adaa449740efa4a8a0f2383d58b1e9106545f2/pkg/lint/linter/config.go#L37-L39

https://github.com/golangci/golangci-lint/blob/92adaa449740efa4a8a0f2383d58b1e9106545f2/pkg/lint/linter/config.go#L41-L50

So all the linters are always considered as fast except `unused` because the status is hard-coded (`ConsiderSlow()`)

https://github.com/golangci/golangci-lint/blob/92adaa449740efa4a8a0f2383d58b1e9106545f2/pkg/lint/lintersdb/manager.go#L151-L157

`packages.NeedTypesInfo` has been removed inside the PR #758 about CPU and memory optimization.

There is no "replacement" for `packages.NeedTypesInfo`: we cannot use another load mode because there is no real relation between the current load modes and the speed.
For example `goheader` or `asciicheck` use the `WithLoadForGoAnalysis` but they are fast.

So, first I will remove the conditions related to the load mode in `IsSlowLinter` because they are useless.

After I don't know what is the best way:
- add `ConsiderSlow()` for all "slow" linters (it will be difficult to maintain)
- remove the `fast` option
- another way?
